yarn-project/simulator/src/avm/errors.ts

```diff
@@ -39,6 +40,37 @@ export class InvalidProgramCounterError extends AvmExecutionError {
   }
 }
 
+/**
+ * Error is thrown when the program counter points to a byte
+ * of an invalid opcode.
+ */
+export class InvalidOpcodeError extends AvmExecutionError {
+  constructor(opcode: Opcode) {
+    super(`Opcode ${Opcode[opcode]} (0x${opcode.toString(16)}) not implemented`);
```

Comment:
BTW, I know this was there already but it's not ideal. It works ok if we have an opcode in the list that is not implemented, but it does not work if you encounter a value which is not any of the existing opcodes (and you'd want to catch that as well).